app/plugins/Hosting/js/hosting.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+// Library for communicating with Sia-UI
+const IPC = require('ipc');
+// Library for arbitrary precision in numbers
+const BigNumber = require('bignumber.js');
+// Variables to store api result values
+var balance = 0;
+// Keeps track of if the view is shown
+var updating;
+
+var view, ePropBlueprint, eProps, eControl, eSave, eReset, eAnnounce;
+var eContracts, eStorage, eRemaining, eProfit, ePotentialProfit;
+var updateTime = 0;
+
+var hostProperties = [
+    {
+        "name": "TotalStorage",
+        "unit": "GB",
+        "conversion": 1/1e9
+    },{
+        "name": "MaxFilesize",
+        "unit": "MB",
+        "conversion": 1/1e6
+    },{
+        "name": "MaxDuration",
+        "unit": "Day",
+        "conversion": 1/144
+    },{
+        "name": "Price",
+        "unit": "SC Per GB Per Month",
+        "conversion": 4/1e12
+    }
+];
+
+// TODO: don't generate these, just use hostProperties
+var editableProps = hostProperties.map(function(obj){
+    return obj["name"];
+});
+var propUnits = hostProperties.map(function(obj){
+    return obj["unit"];
+});
+var propConversion = hostProperties.map(function(obj){
+    return obj["conversion"];
+});
+var lastHostInfo;
+
+
+function callAPI() {
+}
+
+function formatSiacoin(baseUnits) {
+	var ConversionFactor = new BigNumber(10).pow(24);
+	var display = new BigNumber(baseUnits).dividedBy(ConversionFactor);
+	return display + ' SC';
+}
+
+function init() {
+	// DEVTOOL: uncomment to bring up devtools on plugin view
+	//IPC.sendToHost('devtools');
+	
+	// Ensure precision
+	BigNumber.config({ DECIMAL_PLACES: 24 })
+	BigNumber.config({ EXPONENTIAL_AT: 1e+9 })
+
+    view = $("#hosting");
+
+    ePropBlueprint = view.find(".property.blueprint");
+    eAnnounce = view.find(".announce");
+    eControl = view.find(".control");
+    eProps = $();
+    eSave = view.find(".control .save");
+    eReset = view.find(".control .reset");
+    eContracts = view.find(".contracts");
+    eStorage = view.find(".storage");
+    eRemaining = view.find(".remaining");
+    eProfit = view.find(".profit");
+    ePotentialProfit = view.find(".potentialprofit");
+
+    addEvents();
+	updating = setInterval(updating, 1000)
+}
+
+function kill() {
+	clearInterval(updating);
+}
+
+function addEvents(){
+    eAnnounce.click(function(){
```

Comment:
It's actually a naming convention from old Sia-UI, all "eXxxxx" variables point to DOM elements (mostly found through jquery)